### PR TITLE
[HttpClient & HttpWebRequest] Set `http.request.method` as per spec

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Towards #4994 
Design discussion issue #

## Changes

Fixes `http.request.method` attribute value on activity and metric as per spec. For details, see the linked issue

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by TimothyMothra — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs [hunk 1]
```diff
@@ -185,7 +186,16 @@ public void OnStartActivity(Activity activity, object payload)
             // see the spec https://github.com/open-telemetry/semantic-conventions/blob/v1.21.0/docs/http/http-spans.md
             if (this.emitNewAttributes)
             {
-                activity.SetTag(SemanticConventions.AttributeHttpRequestMethod, HttpTagHelper.GetNameForHttpMethod(request.Method));
+                if (RequestMethodHelper.TryResolveHttpMethod(request.Method.Method, out var httpMethod))
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs:189]: This code is very confusing to read. It is a "Try" method which returns true or false however both branches use the result as the tag. You have to go and read the code to understand that a false return actually means "_OTHER" was returned.

How could we improve this with readability/maintainability in mind?

**Review by CodeBlanch — APPROVED** [review]
Some feedback about the helper API being confusing but LGTM

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs [hunk 2]
```diff
@@ -368,6 +372,135 @@ public async Task ExportsSpansCreatedForRetries()
         Assert.NotEqual(spanid2, spanid3);
     }
 
+    [Theory]
+    [InlineData("CONNECT", "CONNECT")]
+    [InlineData("DELETE", "DELETE")]
+    [InlineData("GET", "GET")]
```
Comment (utpilla) [inline test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs:378]: Same feedback as https://github.com/open-telemetry/opentelemetry-dotnet/pull/5001#discussion_r1377900722

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs (hunk 1 above)
Comment (TimothyMothra) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs:189]: Same. At first glance I was expecting it to be `null` and had to check the implementation.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs]: fixed.

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs (hunk 1 above)
Comment (CodeBlanch) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs]: I think what you did is better, but now it is violating DRY and causing the "_OTHER" logic to be leaked/duplicated in multiple spots.

Why not something like...

```csharp
public (string resolvedMethod, bool isKnown) ResolveHttpMethod(string originalMethod) {}
```

Used like this...

```csharp
                (string resolvedMethod, bool isKnown) = RequestMethodHelper.ResolveHttpMethod(request.Method);

                activity.SetTag(SemanticConventions.AttributeHttpRequestMethod, resolvedMethod);
                if (!isKnown)
                {
                    activity.SetTag(SemanticConventions.AttributeHttpRequestMethodOriginal, request.Method);
                }
```

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs]: Good point. I will update it.

**Pull request merged**